src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -2624,7 +2624,7 @@ private void EmitDefaultValue(TypeSymbol type, bool used, SyntaxNode syntaxNode)
             if (used)
             {
                 // default type parameter values must be emitted as 'initobj' regardless of constraints
-                if (!type.IsTypeParameter())
+                if (!type.IsTypeParameter() && type.SpecialType != SpecialType.System_Decimal)
```

Comment:
I could be mistaken, but I don't think we can load decimal constants the same way that we load any other constants.

For example, the IL for `decimal x = 4.1m;` is 

```
        IL_0003: ldc.i4.s 41          // Push num onto the stack as int32, short form
        IL_0005: ldc.i4.0             // Push 0 onto the stack as int32
        IL_0006: ldc.i4.0             // Push 0 onto the stack as int32
        IL_0007: ldc.i4.0             // Push 0 onto the stack as int32
        IL_0008: ldc.i4.1             // Push 1 onto the stack as int32
        IL_0009: call instance void [mscorlib]System.Decimal::.ctor(int32, int32, int32, bool, uint8) // Call method indicated on the stack with arguments

```